spark/src/main/java/org/apache/iceberg/spark/source/RowDataRewriter.java

```diff
@@ -103,8 +103,8 @@ public RowDataRewriter(Table table, PartitionSpec spec, boolean caseSensitive,
     OutputFileFactory fileFactory = new OutputFileFactory(
         spec, format, locations, io.value(), encryptionManager.value(), partitionId, taskId);
 
-    TaskWriter<InternalRow> writer;
-    if (spec.fields().isEmpty()) {
+    final TaskWriter<InternalRow> writer;
```

Comment:
We use `final` because it is unlikely that this actually helps. In Java 8, final detection is quite good, which is why non-final variables can be used in closures and lambdas. And final doesn't produce different byte code so it can't do much to help at runtime.

Reply:
This is mostly for my IDE and sanity. I'm not very puristic about the final usage, but in this case, I think it really adds value. If the if/else construction changes below, and it won't be assigned, it will give an explicit error.

Reply:
What IDE? IntelliJ catches it without `finally`:
![warning-cropped](https://user-images.githubusercontent.com/87915/105114230-4fba0380-5a7b-11eb-86af-eb3bee72a896.png)

Reply:
I'm also using IDEA, but I don't get the error. Probably some configuration issue 🤔